File: android/DateCalcualtor/src/jp/teres/numa/termcalc/TermcalcParent.scala
package jp.teres.numa.termcalc

import android.app.{Fragment, FragmentManager}
import android.view._
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.support.v13.app.FragmentPagerAdapter
import android.util.Log

import jp.teres.numa.DateCalculator._
import jp.teres.numa.utils.Conversions._

class TermcalcParent extends Fragment {

	override def onCreateView(inflater : LayoutInflater, container : ViewGroup, savedInstance : Bundle) : View = {
		val pager = new ViewPager(getActivity)
		pager.setId(R.id.viewPager)
		pager
	}

	override def onResume : Unit = {
		super.onResume
		val fragments = StartDate() :: EndDate() :: CalcResult() :: Nil
		val adapter = new TermcalcParentAdapter(getFragmentManager, fragments)
		getActivity.findViewById(R.id.viewPager).asOpt[ViewPager].foreach(_.setAdapter(adapter))
	}

	override def onPause : Unit = {
		super.onPause
		getView.findViewById(R.id.viewPager)
					.asOpt[ViewPager]
					.map(_.getAdapter)
					.collect{case a : TermcalcParentAdapter => a}
					.foreach { a=> 
						val transaction = getFragmentManager.beginTransaction
						for( i <- 0 until a.getCount()) {
							transaction.remove(a.getItem(i))
						}
						transaction.commitAllowingStateLoss
					}
	}
}

private class TermcalcParentAdapter(fm : FragmentManager, fragments : List[Fragment]) extends FragmentPagerAdapter(fm) {
	
	override def getCount : Int = fragments.size

	override def getItem(position: Int) : Fragment = fragments(position)	
}

File: android/app/src/main/java/jp/teres/numa/DateCalculator/jp/teres/numa/tercalc/TermcalcParent.java
package jp.teres.numa.DateCalculator.jp.teres.numa.tercalc;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Arrays;
import java.util.List;

import jp.teres.numa.DateCalculator.R;

public class TermCalcParent extends Fragment{


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final ViewPager viewPager = new ViewPager(getActivity());
        viewPager.setId(R.id.view_pager);
        return viewPager;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        final ViewPager viewPager = (ViewPager)view.findViewById(R.id.view_pager);
        final List<Fragment> fragments = Arrays.asList(SelectDate.createInstance(SelectDate.SelectDateArgs.Start),
                                                       SelectDate.createInstance(SelectDate.SelectDateArgs.End),
                                                       new CalcResult());

        final FragmentPagerAdapter adapter = new TermcalcParentAdapter(getChildFragmentManager(), fragments);
        viewPager.setAdapter(adapter);
    }

    private static class TermcalcParentAdapter extends FragmentPagerAdapter {

        final List<Fragment> fragments;

        public TermcalcParentAdapter(FragmentManager fragmentManager, List<Fragment> fragments) {
            super(fragmentManager);
            this.fragments = fragments;
        }

        @Override
        public Fragment getItem(int position) {
            return fragments.get(position);
        }

        @Override
        public int getCount() {
            return fragments.size();
        }
    }
}
